hashring/checksummer.go

```diff
@@ -35,22 +36,30 @@ func (i *identityChecksummer) Checksum(ring *HashRing) uint32 {
 	}
 
 	sort.Strings(identities)
-	bytes := []byte(strings.Join(identities, ";"))
-	return farm.Fingerprint32(bytes)
+	i.buf.Reset()
+	for idi, id := range identities {
+		if idi > 0 {
+			i.buf.WriteByte(';')
+		}
+		i.buf.WriteString(id)
+	}
+	return farm.Fingerprint32(i.buf.Bytes())
 }
 
-type replicaPointChecksummer struct{}
+type replicaPointChecksummer struct {
```

Comment:
this change the types unsafe when there's concurrency. are we sure they're not used from multiple goroutines? can we add a large note too